// ngl.js/ngl.ts
import axios, { 
  AxiosRequestConfig 
} from 'axios';

interface Proxy {
  host: string;
  port: number;
  auth: {
    username: string;
    password: string;
  };
};

class NGL {
  private url: string;
  private xurl: string;
  private proxy?: Proxy;

  constructor(proxy?: Proxy) {
    this.url = "https://ngl.link/";
    this.xurl = this.url + "/api/submit";

    if (proxy) {
      this.proxy = proxy;
    };
  }

  async sendMessage(
    username: string,
    message: string
  ): Promise<any> {
    const payload = {
      username: username,
      question: message
    };
    const headers = {
      'Content-Type': 'application/json'
    };

    const config: AxiosRequestConfig = {
      headers
    };

    if (this.proxy) {
      config.proxy = {
        host: this.proxy.host,
        port: this.proxy.port,
        auth: this.proxy.auth
      };
    };

    try {
      const res = await axios.post(
        this.xurl, payload, config
      );

      if (res.status === 200) {
        return res.data;
      } else {
        throw new Error("Failed to send message :(");
      }
    } catch (err: any) {
      throw new Error(err.message);
    }
  }
}

export { NGL };
